### DISCUSSION TITLE
Run on all XScreens?

### DISCUSSION THREAD

**Discussion by ghost** [leftwm/leftwm#1113] [Q&A] [answered]
I gave this a try and found it only functions on XScreen 0.0. I tried launching a separate instance per XScreen to the same result.

**Comment by Eskaan** [comment]
So, to make sure we are taking about the same thing, do you mean the XScreen that is specified using the `DISPLAY` env variable? And so you are talking about leftwm running on `DISPLAY=:0` by default?

I think you can specify the display to run on with the mentioned env variable like this: `DISPLAY=:1 leftwm` (example for display `:1`). You probably will have to make sure that your theme can cope with it. I assume on the side of LeftWM itself the state file should be broken, but that is only relevant for reloads of the WM. Otherwise LeftWM should run just fine.

**Reply by ghost** [reply]
DISPLAY=:1 is a Gnome only Syntax that no longer works. GTK since around version 3.2 is broken for this and GTK hard codes XScreen enumeration to 1 and only 1.

As I said I tried simply launching leftwm and then tried launching a separate instance per XScreen i.e.
DISPLAY=:0.0 leftwm &
DISPLAY=:0.1 leftwm &

However it only runs on 0.0 leaving everything on any other XScreen "there" but you can't interact with anything, no input.

There was a time when DISPLAY=:0 would "encapsulate" the X root but that was like 20 years ago.

**Reply by Eskaan** [reply]
Could you look into your logs and see if leftwm prints any warns or error messages?
Probably `journalctl -t leftwm-worker -p error..warn`

**Comment by ghost** [comment, marked as answer]
Not required, I went back and looked at things to get the log entries you wanted and noticed my .xinitrc was stalling on the first XScreen. An Ampersand later and it does indeed work on all XScreens so long as I launch a separate instance for each. Thanks for replying and getting me to take another look.
